### PR TITLE
add grape_extenders

### PR DESCRIPTION
**Pull request by fgv02009**
@bellycard/platform 

related discussion: https://github.com/bellycard/core-service/pull/32

### CODE REVIEW

**Comment by shaqq** [comment]
👍  looks great, thanks @fgv02009

File: lib/napa/grape_extensions/grape_extenders.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        # if AR is being used, rescue from common AR errors
+        if defined?(::ActiveRecord)
+          modified_class.rescue_from ::ActiveRecord::RecordNotFound do
+            err = Napa::JsonError.new(:record_not_found, 'record not found')
+            Rails.logger.debug({ status: 404, headers: {}, response: err.try(:first) }.map { |k, v| "#{k}=#{v}" }.join(' '))
+            rack_response(err.to_json, 404)
+          end
+          modified_class.rescue_from ::ActiveRecord::RecordInvalid do |e|
+            err = Napa::JsonError.new(:unprocessable_entity, e.message, e.record.errors.messages)
+            Napa::Logger.logger.debug(Napa::Logger.response(422, {}, err))
```
Comment (itsbagpack) [inline lib/napa/grape_extensions/grape_extenders.rb]: Should this be using `Rails.logger` as well?

**Comment by fgv02009** [comment]
updated. bump

File: lib/napa/grape_extensions/grape_extenders.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            Rails.logger.debug({ status: 422, headers: {}, response: err.try(:first) }.map { |k, v| "#{k}=#{v}" }.join(' '))
+            rack_response(err.to_json, 422)
+          end
+        end
+
+        # if AASM is being used, rescue from invalid transitions
+        if defined?(::AASM)
+          modified_class.rescue_from ::AASM::InvalidTransition do |e|
+            err = Napa::JsonError.new(:unprocessable_entity, e.message)
+            Rails.logger.debug({ status: 422, headers: {}, response: err.try(:first) }.map { |k, v| "#{k}=#{v}" }.join(' '))
```
Comment (itsbagpack) [inline lib/napa/grape_extensions/grape_extenders.rb:25]: /nitpick Worth pulling this out into a method? Seem to be using the same pattern at least three times.

**Comment by itsbagpack** [comment]
One nitpick. +1 otherwise.

File: lib/napa/grape_extensions/grape_extenders.rb (hunk 2 above)
Comment (fgv02009) [inline lib/napa/grape_extensions/grape_extenders.rb:25]: Talked to yumin about this offline. Not as east as it seems as `self` here is actually a `Grape::Middleware::Error` going to leave as is.

**Pull request merged**